docs/design/model-deployment.md

```diff
@@ -0,0 +1,200 @@
+---
+id: model-deployment
+title: Model Deployment (Alpha)
+---
+
+Allow users to deploy a model as a service. 
+
+
+## Features
+
+1. **Multiple ML framework support:** Supports Tensorflow 1, Tensorflow 2, Keras, Pytorch, XGBoost, MXNet, Scikit-learn, LightGBM
+1. **Multiple language support:** Supports Python, Java, R, NodeJS, Go (Please see the Seldon wrapper in the Seldon wrapper document)
+1. **Horizontal scaleout:** The deployed model service can be scaled to multiple replicas of the model service. So that it can achieve load balancing and fault tolerance easily.
+1. **Deployment history:** Track the deployment history.
+1. **Resource constraint:** The usage of resources for model deployment is constrained by group resource quota.
+1. **Ingress:** Create the ingress resource to redirect external requests to internal model service.
+
+## User Journey
+
+Deploy a model
+- (Admin) Enable a model deployment for one group
+- Create a deployment. Select the instance type, model image, and the number of replicas.
+- Wait until the deployment is ready
+- Use `curl` to test against the model deployment endpoint
+
+Package a model 
+- Train a model for some ML framework and select the best model for deployment
+- Wrap the model file and build the image by `s2i`
+- Test the packaged image locally
+- Push the image to docker registry
+
+Update a deployment
+- Select a deployment
+- Click the Update button
+- Change the image and deploy
+
+# Configuration
+
+Please add this variable to the `.env` file. 
+
+Name | Value 
+--- | ----- 
+`PRIMEHUB_FEATURE_MODEL_DEPLOYEMNT` | `true`
+
+Chart value
+
+Path | Description | Default Value 
+--- | ----- | -----------------------
+`modelDeployment.enabled` | Enable the model deployment | `false`
+
+# Design
+
+## Seldon
+
+[Seldon](https://www.seldon.io/) is a model deployment solution in the community. The reason why we select Seldon as the solution is because it provides a common way to package different framework's by different programming languages into a docker image.
+
+Seldon also provides an operator under `Seldon core` project to manage a `SeldonDeployment` resource and reconcile it to the underlying deployments and services. However, for simplicity, we decide not to use the `SeldonDeployment` resource. Instead, we define `PhDeployment` and the controller generate the underlying `Deployment` and `Service` directly.
```

Comment:
Doesn't sound grammatically correct to me...
- Instead, we define `PhDeployment` **along with** the controller **and** generate the underlying `Deployment` and `Service` directly.
- Instead, we define `PhDeployment` and the controller **for generating/to generate** the underlying `Deployment` and `Service` directly.

Reply:
Fixed